runtime/runtime/src/actions.rs

```diff
@@ -685,20 +684,27 @@ pub(crate) fn action_use_global_contract(
         .into());
         return Ok(());
     }
-    if let AccountContract::Local(code_hash) = account.contract().as_ref() {
-        let prev_code_len = get_code_len_or_default(
-            state_update,
-            account_id.clone(),
-            *code_hash,
-            current_protocol_version,
-        )?;
-        account.set_storage_usage(account.storage_usage().saturating_sub(prev_code_len));
-        state_update.remove(TrieKey::ContractCode { account_id: account_id.clone() });
-    }
+    clear_account_contract_storage_usage(
+        state_update,
+        account_id,
+        account,
+        current_protocol_version,
+    )?;
+    state_update.remove(TrieKey::ContractCode { account_id: account_id.clone() });
```

Comment:
this should only be executed under `if let AccountContract::Local(code_hash) = account.contract().as_ref()`